### ISSUE TITLE
`doc` should report errors for mangled types

### ISSUE THREAD

**Issue by sanbox-irl** [rust-lang/rust#141095]
### Problem

I wrote:

```
[\`TextManager:atlases\`]
```

but I meant to write `TextManager::atlases`

The error I got was:

```
warning: unresolved link to `TextManager:atlases`
```

But a better error (for my case) would have been referencing that I was missing a semicolon.

### Proposed Solution

Since I believe *any* item cannot have a single `:` in it, I propose the error should just lint for any illegal characters in a path.

I understand this might be more complex because of partial paths, but I am not familiar enough with those to address them here.
